lib/kino/explorer.ex

```diff
@@ -38,31 +38,52 @@ defmodule Kino.Explorer do
         }
       end)
 
-    info = %{name: "DataFrame", features: [:pagination, :sorting]}
+    info = %{name: "DataFrame", features: [:pagination, :sorting, :filtering]}
 
     {:ok, info, %{df: df, total_rows: total_rows, columns: columns}}
   end
 
   @impl true
   def get_data(rows_spec, state) do
-    records = get_records(state.df, rows_spec)
+    {records, total_rows, summaries} = get_records(state.df, rows_spec)
+    columns = Enum.map(state.columns, &%{&1 | summary: summaries[&1.key]})
     rows = Enum.map(records, &record_to_row/1)
-    {:ok, %{columns: state.columns, rows: rows, total_rows: state.total_rows}, state}
+    {:ok, %{columns: columns, rows: rows, total_rows: total_rows}, state}
   end
 
   defp get_records(df, rows_spec) do
     df =
-      if order_by = rows_spec[:order_by] do
-        Explorer.DataFrame.arrange_with(df, &[{rows_spec.order, &1[order_by]}])
-      else
-        df
-      end
+      df
+      |> order_by(rows_spec.order, rows_spec[:order_by])
+      |> filter_by(rows_spec.filter, rows_spec[:filter_by], rows_spec.filter_value)
 
+    total_rows = Explorer.DataFrame.n_rows(df)
+    summaries = if total_rows > 0, do: summaries(df)
     df = Explorer.DataFrame.slice(df, rows_spec.offset, rows_spec.limit)
-
     {col_names, lists} = df |> Explorer.DataFrame.to_columns() |> Enum.unzip()
+    records = Enum.zip_with(lists, fn row -> Enum.zip(col_names, row) end)
+    {records, total_rows, summaries}
+  end
+
+  defp order_by(df, order, order_by) do
+    if order_by, do: Explorer.DataFrame.arrange_with(df, &[{order, &1[order_by]}]), else: df
+  end
```

Comment:
wdyt? same for `filter_by` :)

```suggestion

  defp order_by(df, _order, nil), do: df
  
  defp order_by(df, order, order_by) do
    Explorer.DataFrame.arrange_with(df, &[{order, &1[order_by]}])
  end
```